### PR TITLE
feat: Export ECS container logs to Grafana Loki

### PR DESCRIPTION
**Pull request by fabiolrodriguez**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -92,6 +92,12 @@ npm start
 
 See: https://docs.sst.dev for more info on how things get deployed.
 
+To remove dev services yo our aws accout:
```
Comment (blucas) [inline README.md:95]:
```suggestion
To remove dev services your aws account:
```

File: stacks/PickupStack.ts [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
 import { QueueProcessingFargateService } from './lib/queue-processing-fargate-service'
+import { Group, ManagedPolicy } from 'aws-cdk-lib/aws-iam'
+import { servicesVersion } from 'typescript'
+import { urlSource } from 'ipfs/dist/src'
+import { aws_secretsmanager, SecretValue } from 'aws-cdk-lib'
 
 export function PickupStack ({ stack }: StackContext): void {
   const basicApi = use(BasicApiStack) as unknown as { queue: Queue, bucket: Bucket }
-
+
```
Comment (blucas) [inline stacks/PickupStack.ts:13]:
```suggestion

```

File: stacks/PickupStack.ts [hunk 3]
```diff
@@ -30,11 +34,35 @@ export function PickupStack ({ stack }: StackContext): void {
...
+      type: FirelensLogRouterType.FLUENTBIT,
+    },
+    image: ContainerImage.fromRegistry('grafana/fluent-bit-plugin-loki:1.6.0-amd64'),
+  })
+
+  const grafanasecret = aws_secretsmanager.Secret.fromSecretNameV2(
+    stack,
+    'gf-id',
+    'grafanahost',
+  );
```
Comment (blucas) [inline stacks/PickupStack.ts:50]: We will want to make exporting logs to loki conditional per environment.  For example, PR based envs should not be sending that info over.

File: stacks/PickupStack.ts [hunk 4]
```diff
@@ -30,11 +34,35 @@ export function PickupStack ({ stack }: StackContext): void {
...
+    logging: LogDrivers.firelens({    
+      options: {
+        Name: "loki",
+        labels: "{job=\"pickup_production\"}",
+        remove_keys: "container_id,ecs_task_arn",
+        label_keys: "container_name,ecs_task_definition,source,ecs_cluster",
+        line_format: "key_value",
+        url: grafanasecret.secretValue.toString()
+      },
+    }),
```
Comment (blucas) [inline stacks/PickupStack.ts:65]: Conditional per environment

File: stacks/PickupStack.ts [hunk 5]
```diff
@@ -50,3 +78,7 @@ export function PickupStack ({ stack }: StackContext): void {
   basicApi.bucket.cdk.bucket.grantReadWrite(service.taskDefinition.taskRole)
   basicApi.queue.cdk.queue.grantConsumeMessages(service.taskDefinition.taskRole)
 }
+
+
+
+
```
Comment (blucas) [inline stacks/PickupStack.ts:84]:
```suggestion
```

**Review by blucas — CHANGES_REQUESTED** [review]

File: stacks/PickupStack.ts (hunk 3 above)
Comment (fabiolrodriguez) [reply stacks/PickupStack.ts]: Now the job name is conditional, based on the env name, like staging-pickup, prod-pickup or pr58-pickp (this pr)

image

File: stacks/PickupStack.ts [hunk 6]
```diff
@@ -1,12 +1,16 @@
...
+import { ContainerImage, LogDriver, AwsLogDriver, LogDrivers, FireLensLogDriver, Secret, FirelensLogRouter, FirelensLogRouterProps, FirelensLogRouterType, Scope} from 'aws-cdk-lib/aws-ecs'
 import { Platform } from 'aws-cdk-lib/aws-ecr-assets'
 import { QueueProcessingFargateService } from './lib/queue-processing-fargate-service'
+import { Group, ManagedPolicy } from 'aws-cdk-lib/aws-iam'
+import { servicesVersion } from 'typescript'
+import { urlSource } from 'ipfs/dist/src'
+import { aws_ssm } from 'aws-cdk-lib'
+import { aws_secretsmanager, SecretValue } from 'aws-cdk-lib'
 
 export function PickupStack ({ stack }: StackContext): void {
```
Comment (blucas) [inline stacks/PickupStack.ts:12]:
```suggestion
export function PickupStack ({ app, stack }: StackContext): void {
```

File: stacks/PickupStack.ts [hunk 7]
```diff
@@ -31,21 +35,60 @@ export function PickupStack ({ stack }: StackContext): void {
     enableExecuteCommand: true
   })
 
+  var labelname = new String(stack);
+  labelname = labelname.slice(0, -12)
+
+    if (labelname == "prod-pickup" || labelname == "staging-pickup") {
```
Comment (blucas) [inline stacks/PickupStack.ts:41]:
```suggestion
    if (app.stage in ['prod', 'staging']) {
```

**Review by blucas — APPROVED** [review]

**Pull request merged**